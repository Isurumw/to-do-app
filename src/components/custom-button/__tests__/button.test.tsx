import 'react-native';
import React from 'react';

import Button from '../button';
import {render, fireEvent} from 'react-native-testing-library';

describe('check the behaviour of the custom button component', () => {
  it('the custom button component renders correctly', () => {
    const component = render(<Button label="SIGN IN" onPress={jest.fn()} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('press the button', () => {
    const mockHandleButtonPress: jest.Mock<void> = jest.fn();

    const component = render(
      <Button label="SIGN IN" onPress={mockHandleButtonPress} />,
    );

    const button = component.getByTestId('custom_button');

    fireEvent.press(button);
    expect(mockHandleButtonPress).toBeCalledTimes(1);
  });
});
