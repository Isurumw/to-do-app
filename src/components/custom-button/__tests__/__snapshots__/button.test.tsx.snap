// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the behaviour of the custom button component the custom button component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#7C2AF7",
      "borderRadius": 50,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 12,
    }
  }
  testID="custom_button"
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "DMSans-Medium",
        "fontSize": 13,
        "letterSpacing": 0.02,
        "lineHeight": 16,
      }
    }
  >
    SIGN IN
  </Text>
</View>
`;
